### PR TITLE
fix(add): exact version should not have range `^` specifier

### PR DESCRIPTION
**Pull request by marvinhagemeister**
Fixes https://github.com/denoland/deno/issues/26299

### CODE REVIEW

**Comment by marvinhagemeister** [comment]
oh wait it looks like the same issue also exists with JSR specifiers. Marking as `draft` again.

**Comment by marvinhagemeister** [comment]
Hm looks like checking the `VersionReq` afterwards is not the right move, as they will also be the same when you specify no version and there is only one version that matches.

**Review by dsherret — APPROVED** [review]
LGTM!

File: cli/tools/registry/pm.rs [hunk 1]
```diff
@@ -593,6 +594,20 @@ enum PackageAndVersion {
   Selected(SelectedPackage),
 }
 
+fn is_exact_version_req(req: &PackageReq) -> bool {
+  match req.version_req.inner() {
+    RangeSetOrTag::RangeSet(range_set) => {
+      let items = &range_set.0;
+      if let Some(first) = items.first() {
+        first.start == first.end && items.len() == 1
```
Comment (dsherret) [inline cli/tools/registry/pm.rs:602]: Oh, I just noticed the failing test and comment. I think you can check the `RangeBound`, but I'm seeing we accidentally don't export it from deno_semver: https://github.com/denoland/deno_semver/pull/37

**Review by dsherret — APPROVED** [review]
LGTM!

**Pull request merged**